src/Corax/IndexEntry/IndexEntryWriter.cs

```diff
@@ -91,236 +117,254 @@ public void WriteRaw(int field, ReadOnlySpan<byte> binaryValue)
+            dataLocation += value.Length;
+        }
+        else
+        {
+            // Signal that we will have to deal with the empties.
+            indexEntryField |= IndexEntryFieldType.Empty;
         }
+
+        _dataIndex = dataLocation;
     }
 
-    public void Write<TEnumerator>(int field, TEnumerator values, IndexEntryFieldType type = IndexEntryFieldType.Null)
+    public void Write<TEnumerator>(int field, TEnumerator values, IndexEntryFieldType type = IndexEntryFieldType.Null) 
         where TEnumerator : IReadOnlySpanIndexer
     {
         Debug.Assert(field < _knownFields.Count);
-        Debug.Assert(_knownFieldsLocations[field] == Invalid);
+        Debug.Assert(KnownFieldsLocations[field] == Invalid);
+
+        // Calculate the required space. 
+        int requiredSpace = 0;
+        for (int i = 0; i < values.Length; i++)
+            requiredSpace += values[i].Length;
+        requiredSpace += sizeof(IndexEntryFieldType) + 2 * values.Length * sizeof(long) + 4 * sizeof(long);
+        
+        if (FreeSpace < requiredSpace)
+            UnlikelyGrowAuxiliaryBuffer(requiredSpace);
 
         int dataLocation = _dataIndex;
+        var buffer = Buffer;
 
         // Write known field pointer.
-        ref int fieldLocation = ref _knownFieldsLocations[field];
+        ref int fieldLocation = ref KnownFieldsLocations[field];
         fieldLocation = dataLocation | Constants.IndexWriter.IntKnownFieldMask;
 
         // Write the list metadata information. 
-        fixed (byte* bufferPtr = _buffer)
-        {
-            ref var indexEntryField = ref Unsafe.AsRef<IndexEntryFieldType>(bufferPtr + dataLocation);
+        ref var indexEntryField = ref Unsafe.AsRef<IndexEntryFieldType>(Unsafe.AsPointer(ref buffer[dataLocation]));
+        indexEntryField = IndexEntryFieldType.List | type;
+        dataLocation += sizeof(IndexEntryFieldType);
 
-            indexEntryField = IndexEntryFieldType.List | type;
-            dataLocation += sizeof(IndexEntryFieldType);
+        // Size of list.
+        dataLocation += VariableSizeEncoding.Write(buffer, values.Length, dataLocation);
 
-            // Size of list.
-            dataLocation += VariableSizeEncoding.Write(_buffer, values.Length, dataLocation);
-
-            // Prepare the location to store the pointer where the table of the strings will be (after writing the strings).
-            ref int stringPtrTableLocation = ref Unsafe.AsRef<int>(bufferPtr + dataLocation);
-            dataLocation += sizeof(int);
-
-            // Copy the actual string data. 
-            if (values.Length == 0)
-            {
-                // Write the pointer to the location.
-                stringPtrTableLocation = Invalid;
+        // Prepare the location to store the pointer where the table of the strings will be (after writing the strings).
+        ref int stringPtrTableLocation = ref Unsafe.AsRef<int>(Unsafe.AsPointer(ref buffer[dataLocation]));
```

Comment:
avoid this

Reply:
Same as in above comment about avoiding this

Reply:
In the case of `IndexEntryWriter` we own the memory and we know it comes from unmanaged; so we can do that without any required usage of `fixed`. We use `Span<byte>` for convenience and bound check semantics; but we could have very well used the pointers directly.